src/test/rpc/AccountTx_test.cpp

```diff
@@ -166,11 +166,21 @@ class AccountTx_test : public beast::unit_test::suite
 
             p[jss::ledger_index_min] = 0;
             p[jss::ledger_index_max] = 100;
-            BEAST_EXPECT(hasTxs(env.rpc("json", "account_tx", to_string(p))));
+            RPC::apiMaximumSupportedVersion == 1
+                ? BEAST_EXPECT(
+                      hasTxs(env.rpc("json", "account_tx", to_string(p))))
+                : BEAST_EXPECT(isErr(
+                      env.rpc("json", "account_tx", to_string(p)),
+                      rpcLGR_IDX_MALFORMED));
```

Comment:
You should copy these tests for an explicit `api_version == 1`, like [this](https://github.com/XRPLF/rippled/pull/4552/files#diff-d49d692a95cf0496bc5f953d37f2535730d925b6ddce2a0eb004f2f6ecef6f05R1232-R1256).

Reply:
To follow on @thejohnfreeman's suggestion, loop over both versions for these tests, because we need to keep supporting version 1 indefinitely and ensure it's working correctly.